### PR TITLE
Remove Chewy::Type, simplify DSL

### PR DESCRIPTION
**Pull request by rabotyaga**
Remove the type concept from internals completely, as Elasticsearch is removing types:
- Remove the `Chewy::Type` class
  - e.g. remove `CitiesIndex::City` / `CitiesIndex.city`
    - `CitiesIndex::City.import! ...` becomes `CitiesIndex.import! ...`
- Simplify index DSL:
  - `define_type` block -> `index_scope` clause
    - it can be omitted completely, if you don't need to specify the scope or options, e.g. `name`
- Remove type names from string representations:
  - in `update_index` ActiveRecord helper and RSpec matcher, e.g.
    - `update_index('cities#city')` -> `update_index('cities')`
    - `update_index(UsersIndex::User)` -> `update_index(UsersIndex)`
  - in rake tasks (e.g. `rake chewy:update[cities#city]` -> `rake chewy:update[cities]`)
  - in rake tasks output (e.g. `Imported CitiesIndex::City in 1s, stats: index 3` -> `Imported CitiesIndex in 1s, stats: index 3`)
- Use index name instead of type name in loader additional scope
  - e.g. `CitiesIndex.filter(...).load(city: {scope: City.where(...)})` -> `CitiesIndex.filter(...).load(cities: {scope: City.where(...)})`

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the changelog if the new code introduces user-observable changes. See [changelog entry format](https://github.com/toptal/chewy/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Review by AlfonsoUceda — APPROVED** [review]

File: lib/chewy/index.rb [hunk 1]
```diff
@@ -113,69 +138,42 @@ def prefix
...
       #   end
       #
       # Type name might be passed in complicated cases:
       #
-      #   class VehiclesIndex < Chewy::Index
-      #     define_type Vehicle.cars.includes(:manufacturer), name: 'cars' do
-      #        ...
-      #     end # defines VehiclesIndex::Cars type
+      #   class CarsIndex < Chewy::Index
+      #     index_scope Vehicle.cars.includes(:manufacturer), name: 'cars'
```
[the remark this answers is not in the document]
Comment (rabotyaga) [reply lib/chewy/index.rb:152]: Sorry?

[the remark this answers is not in the document]
Comment (rabotyaga) [reply lib/chewy/index.rb]: I think we can remove this part of doc completely

[the remark this answers is not in the document]
Comment (AlfonsoUceda) [reply lib/chewy/index.rb]: Don't worry, I think you answered my question in another place.

After reading the question, I think I wrote it wrong

**Review by dalthon — APPROVED** [review]
Assuming that most of the work was based to move modules that once were included at `Chewy::Type` to `Chewy::Index`, and changing edge cases like `index_scope`, it LGTM!

**Pull request merged**